File: Easy/PlusOne.cpp
#include<iostream>
#include<vector>

using namespace std;

vector<int> plusOne(vector<int>& digits) {
    int last;
    int sum = 0;

    for(int e : digits)
        sum = sum*10 + e;
    sum++;

    vector<int> p1;

    if(sum == 10){
        digits.pop_back();
        digits.push_back(1);
        digits.push_back(0);
    }
    else{
        last = sum%10;
        digits.pop_back();
        digits.push_back(last);
    }
    return digits;
}


int main(){
    vector<int> v{9};
    auto res = plusOne(v);
    cout<<"Result: ";
    for (auto i : res)
        cout << i << " ";
}

File: Easy/PlusOne.py
def plusOne(digits: list[int]) -> list[int]:
    d = 0
    for digit in digits:
        d = d*10+digit 
    d += 1    
    digits.clear()
    
    while d>0:
        last = d%10
        digits.insert(0, last)
        d //= 10

    return digits
print(plusOne([1]))